### PR TITLE
add Q_ENUM macro to QgsVectorLayer::EditResult and SelectBehavior (and fix master build)

### PR DESCRIPTION
**Pull request by 3nids**
useful for QML

### CODE REVIEW

**Review by m-kuhn — APPROVED** [review]
Looks good to me
And thanks for the proj6 build fix

File: src/3d/CMakeLists.txt [hunk 1]
```diff
@@ -165,7 +165,7 @@ IF(MSVC)
   SET_TARGET_PROPERTIES(qgis_3d PROPERTIES LINK_FLAGS "/FORCE:MULTIPLE")
 ENDIF(MSVC)
 
-SET(QGIS_3D_HDRS ${QGIS_3D_HDRS} ${CMAKE_CURRENT_BINARY_DIR}/qgis_3d.h)
+SET(QGIS_3D_HDRS ${QGIS_3D_HDRS} ${QGIS_3D_PRIVATE_HDRS} ${CMAKE_CURRENT_BINARY_DIR}/qgis_3d.h)
```
Comment (m-kuhn) [inline src/3d/CMakeLists.txt]: Don't!!
Move the erroneous ones to the public header section instead.

**Review by m-kuhn — CHANGES_REQUESTED** [review]
Private headers must not be installed.
If some of them need to be installed currently, they cannot be considered private and should go to the public headers instead.
Or better, included in the cpp file instead.

File: src/3d/CMakeLists.txt (hunk 1 above)
Comment (3nids) [reply src/3d/CMakeLists.txt]: ok, done

**Comment by 3nids** [comment]
private headers not installed, only qgsfield_p

**Review by m-kuhn — APPROVED** [review]
Thanks

**Pull request merged**